### ISSUE TITLE
Android Studio Giraffe freezing after run "restart App with debugger"

### ISSUE THREAD

**Issue by NamekMaster** [pbreault/adb-idea#157]
After updated to newest version of Android Studio Giraffe, the IDE is freezing each time I click "restart App with debugger".

Android Studio Info:

```
Android Studio Giraffe | 2022.3.1
Build #AI-223.8836.35.2231.10406996, built on June 29, 2023
Runtime version: 17.0.6+0-17.0.6b829.9-10027231 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.4
GC: G1 Young Generation, G1 Old Generation
Memory: 8192M
Cores: 10
Metal Rendering is ON
Registry:
    editor.focus.mode.color.light=777777
    analyze.exceptions.on.the.fly=true
    ide.editor.tab.selection.animation=true
    external.system.auto.import.disabled=true
    debugger.new.tool.window.layout=true
    ide.text.editor.with.preview.show.floating.toolbar=false
    ide.instant.shutdown=false
    ide.new.editor.tabs.vertical.borders=true
    ide.experimental.ui=true
    ide.balloon.shadow.size=0
    editor.focus.mode.color.dark=777777

Non-Bundled Plugins:
    com.jetbrains.plugins.ini4idea (223.8836.46)
    cn.haojiyou.CodeGlance3 (2.1.1-2021.1.3)
    Quick Notes (3.3)
    com.intellij.javafx (1.0.4)
    Gradle View (4.0.0)
    main.kotlin.com.jimschubert.intellij.swaggercodegen (4.3.1-p0)
    com.github.fawaz.flutter-freezed-snippets (1.0.6)
    Shifter (1.9.6)
    com.kagof.pokeprogress (2.1.1)
    org.jetbrains.settingsRepository (223.8214.6)
    Dart (223.8950)
    com.github.beansoftapp.visualgc.plugin (2022.1.0)
    no.hvl.tk.visualDebugger (1.9.3)
    com.markskelton.one-dark-theme (5.8.0)
    com.godwin.json.parser (1.4.2)
    com.crunch42.openapi (1.52)
    com.intellij.marketplace (223.8836.56)
    String Manipulation (9.9.0)
    com.localizely.flutter-intl (1.18.3-2022.2)
    net.seesharpsoft.intellij.plugins.file-preview (1.6.4)
    org.tbm98.flutter-riverpod-snippets (2.232.1)
    AceJump (3.8.14)
    IdeaVIM (2.1.0)
    IdeaVimExtension (1.6.11)
    com.github.copilot (1.2.15.2816)
    org.jetbrains.IdeaVim-EasyMotion (1.10)
    com.joshestein.ideavim-quickscope (1.0.11)
    com.herbert.george.flutter-snippets (2.0.0-stable-1)
    indent-rainbow.indent-rainbow (2.1.0)
    com.cmgapps.intellij.proguard-retrace-unscambler (1.7.0)
    ski.chrzanow.foldableprojectview (1.1.4)
    tanvd.grazi (223.8836.5)
    PlantUML integration (6.3.0-IJ2022.2)
    PythonCore (223.8836.26)
    org.exbin.deltahex.intellij (0.2.8.1)
    izhangzhihao.rainbow.brackets (2023.3.2)
    io.kanro.idea.plugin.protobuf (1.6.32)
    org.jetbrains.compose.desktop.ide (1.4.3)
    cn.yiiguxing.plugin.translate (3.5.1)
    org.norbye.tor.kdocformatter (1.5.9)
    com.github.grishberg.android.yamp (22.08.10.0)
    com.github.grishberg.android.android-layout-inspector-plugin (23.05.11.0)
    com.jetbrains.kmm (0.6.1(223)-18)
    com.google.d2c.androidstudio.plugin (0.3.07-hotfix)
    com.developerphil.adbidea (1.6.10)
    SequenceDiagram (3.0.5)
    com.vladsch.idea.multimarkdown (3.0.213.150)
    zielu.gittoolbox (500.0.13+213)
    mobi.hsz.idea.gitignore (4.4.4)
    io.flutter (74.0.3)
    de.mariushoefler.flutter_enhancement_suite (1.6.1)
    de.endrullis.idea.postfixtemplates (2.16.2.223)

```

**Comment by twiceyuan** [comment]
Have the same problem.

**Comment by pbreault** [comment]
Thanks for the report. I backported the fix from Hedgehog to Giraffe. 

Fixed in 1.6.12.
